### PR TITLE
Create api

### PR DESCRIPTION
**Pull request by jaskaransarkaria**
- API stores latest release tags for all "cloud-platform-terraform-*" repos in redis
- API routes to get single repo's latest version, all repo's latest versions or update a repo's latest version (the idea is each repo will update this whenever a new version is released)
- POST route is protected
- API populates redis with the most up-to-date versions for all "cloud-platform-terraform-*" repos on startup (unfortunately I couldn't do this in the post start up deployment hook as the HTTP requests kept failing from github)"

To test:

```
curl -i https://modules.apps.live-2.cloud-platform.service.justice.gov.uk/ 
curl -i https://modules.apps.live-2.cloud-platform.service.justice.gov.uk/$REPO_NAME
curl -X POST -H 'X-API-Key: $K8-SECRET' -i https://modules.apps.live-2.cloud-platform.service.justice.gov.uk/update/$REPO_NAME/$VERSION   
```

Next Step: Set up CI/ CD for this repo

### CODE REVIEW

File: deploy/app.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                secretKeyRef:
+                  name: redis-helm
+                  key: redis-password
+            - name: API_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: go-get-module-api-key
+                  key: api_key
+            - name: REDIS_ADDR
+              value: "redis-helm-master.cloud-platform-go-get-module.svc.cluster.local"
```
Comment (jasonBirchall) [inline deploy/app.yaml:38]: Can we make this a call to a config map interpolation?

File: init_app/init_data.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+func InitDataClient(dataAddr, dataPassword string) utils.DataAccessLayer {
+	return initRedis(dataAddr, dataPassword)
+}
+
+func InitData(dataClient utils.DataAccessLayer) {
+	client := github.NewClient(nil)
+	repos, err := getRepos(client)
+
+	if err != nil {
+		log.Println("Error getting repo names: ", err)
```
Comment (jasonBirchall) [inline init_app/init_data.go:23]: Can we return the error from this function and then print the output from the caller?

Comment (jaskaransarkaria) [reply init_app/init_data.go]: Good point I've pushed a fixup.

Comment (jaskaransarkaria) [reply init_app/init_data.go]: https://github.com/ministryofjustice/cloud-platform-go-get-module/pull/2/commits/dd410ed43e9518a9afd586dd7ea4b3427b3d9206

File: deploy/app.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    matchLabels:
+      name: api
+  template:
+    metadata:
+      labels:
+        name: api
+    spec:
+      containers:
+        - name: application
+          image: jaskaransarkaria/cloud-platform-go-get-module:VERSION
```
[the remark this answers is not in the document]
Comment (jaskaransarkaria) [reply deploy/app.yaml:17]: Great point, i've created a pr for the ecr repo in environments. [pr here](https://github.com/ministryofjustice/cloud-platform-environments/pull/10525) I'll change this value once the ecr repo is created.

**Review by jasonBirchall — APPROVED** [review]

**Pull request merged**